src/components/ApplyForm/ApplyForm.jsx

```diff
@@ -1,8 +1,139 @@
 import Button from "../../components/Buttton/Button";
+import { useState } from 'react'
+import { redirect, useNavigate } from 'react-router-dom';
+import Dropdown from "../Dropdown/Dropdown";
+import postCreateAccount from "../../api/post_create_account";
+import postLogin from "../../api/post_login";
 import "./ApplyForm.css";
-import Dropdown from "../../components/Dropdown/Dropdown";
+
 
 const ApplyForm = () =>{
+    const navigate = useNavigate();
+
+
+    const [errorMessage, setErrorMessage] = useState("")
+    const [formInvalid, setFormInvalid] = useState("")
+    const [checkedState, setCheckedState] = useState([]);
+    const [mentored, setMentored] = useState(true);
+
+    // until we have a separate get request for skills list:
+    const skills = [ "Python", "Django", "DRF", "React", "Javascript", "Frontend",  "Backend", "HTML-CSS"]
+
+    const [signupdetails, setSignupDetails] = useState({
+        first_name: "",
+        last_name: "",
+        email: "",
+        mobile: "",
+        social_account: "",
+        linkedin_account: "", 
+        github_profile: "",
+        username: "",
+        password: "",
+        has_mentored:false,
+        location: "",
+        skills: [],
+    })
+
+    const handleChange = (event) => {
+
+            const {id, value} = event.target;
+            setSignupDetails((prevDetails) => ({
+                ...prevDetails,
+                [id]: value,
+            }))
+            
+
+    }
+
+    const handleCheckboxChange = (event) => {
+        let updatedList = [...checkedState]
+        if (event.target.checked) {
+            updatedList = [...checkedState, event.target.value]
+        } else {
+            updatedList.splice(checkedState.indexOf(event.target.value),1)
+        }
+        setCheckedState(updatedList)
+
+
+        setSignupDetails({...signupdetails,
+            skills: updatedList
+    
+        })
+    }
+
+
+
+    const handleSelectionChange =(value) =>{
+        setSignupDetails({...signupdetails,
+        location: value})
+    }
+
+    const handleBooleanChange =(mentored) =>{
+        console.log(mentored)
+        
+        
+        setSignupDetails({...signupdetails,
+        has_mentored:mentored})
+        console.log(signupdetails)
+    }
+
+   
+
+        const handleSubmit = (event) => {
+            console.log(signupdetails)  
+            event.preventDefault()
+    
+      if(
+        signupdetails.first_name &&
+        signupdetails.last_name &&
+        signupdetails.email &&
+        signupdetails.mobile &&
+        signupdetails.social_account &&
```

Comment:
Not all of these details are mandatory. When you attempt to submit without all of them, you get no response or error, it just looks like nothing is working. I would suggest changing this to only include the minimum required fields and have some response if not all required fields are completed.

Reply:
Resolved in https://github.com/SheCodesAus/layer_8_victory_frontend_bris_2023/pull/13/commits/e8d4a0852b43b753a63f2f46c2d71cdab6feba27